auditbeat/module/file_integrity/monitor/monitor_test.go

```diff
@@ -213,16 +309,23 @@ func testDirOps(t *testing.T, dir string, watcher Watcher) {
 var errReadTimeout = errors.New("read timeout")
 
 // helper to read from channel
-func readTimeout(c <-chan fsnotify.Event) (fsnotify.Event, error) {
-	select {
-	case <-time.After(3 * time.Second):
-		return fsnotify.Event{}, errReadTimeout
-
-	case msg, ok := <-c:
-		if !ok {
-			return fsnotify.Event{}, errors.New("channel closed")
+func readTimeout(watcher Watcher) (fsnotify.Event, error) {
+	timer := time.NewTimer(3 * time.Second)
+	defer timer.Stop()
+	for {
+		select {
+		case <-timer.C:
+			return fsnotify.Event{}, errReadTimeout
+
+		case msg, ok := <-watcher.EventChannel():
+			if !ok {
+				return fsnotify.Event{}, errors.New("channel closed")
+			}
+			return msg, nil
+
+		case err := <-watcher.ErrorChannel():
+			fmt.Printf("readTimeout got error: %v\n", err)
```

Comment:
Can you pass a testing.TB interface and use Logf instead. It will keep the output clean for the non-verbose runs.